app/scripts/models/account.js

```diff
@@ -335,8 +362,67 @@ define([
           );
         })
         .then(function (updatedSessionData) {
-          self.set(updatedSessionData);
+          self.storeSessionData(updatedSessionData, relier);
         });
+    },
+
+    /**
+     * Upgrade the data format of an account stored in localStorage, if needed.
+     *
+     * A `storageVersion` field on the account keeps track of which data
+     * format is stored. If no `storageVersion` field exists, the account
+     * is from a pre-versioning period and should be updated to version 1.
+     *
+     * To avoid a combinatoric mess, if data is multiple versions behind,
+     * it must be updated through each version. For example, if the current
+     * version number is 3, and the stored account is unversioned, the
+     * sequence is: unversioned->1->2->3
+     *
+     * Each time the data format changes, two things need to happen:
+     * 1. Bump `CURRENT_STORAGE_VERSION`
+     * 2. Add a new function `upgradeTo<N>` where N is the new value of
+     *    `CURRENT_STORAGE_VERSION`. The function should update data
+     *    formats. The function does not need to update `storageVersion`,
+     *    that is handled automatically.
+     */
+    CURRENT_STORAGE_VERSION: 1,
+    upgradeStorageFormat: function () {
```

Comment:
As far as the account model is concerned, this seems like more of a "data format upgrade" than a "storage format upgrade".